src/compilation/build_module.cc

```diff
@@ -0,0 +1,317 @@
+/*!
```

Comment:
put the file under src/codegen/build_module.cc for now, since for now it is only stich up things and do not need a new module